foxpuppet/windows/browser/tab.py

```diff
@@ -0,0 +1,39 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.region import Region
+
+
+class Tab(Region):
+
+    _tabs_locator = (By.TAG_NAME, 'tab')
+
+    @property
+    def close_button(self):
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tab = self.selenium.find_element(*self._tabs_locator)
+            return tab.find_anonymous_element_by_attribute(
+                'anonid', 'close-button')
+
+    @property
+    def switch_to(self):
+        self.selenium.switch_to.window(self.selenium.current_window_handle)
+
+    def close(self):
+        from foxpuppet.windows.browser.tab_bar import TabBar
+        from foxpuppet.windows.browser.window import BrowserWindow
+
+        window = BrowserWindow(
+            self.selenium, self.selenium.current_window_handle)
+
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tabbar = TabBar(window, Tab)
+            current_tabs = tabbar.tabs
```

Comment:
There's some confusion here. The tabs in the tab bar are not the window handles, but are associated with window handles. As FoxPuppet is for UI modeling, the tab refers to the visual artifact that the user can click/drag, and contains a button that can be clicked to close:

![screenshot 2017-03-10 11 46 09](https://cloud.githubusercontent.com/assets/122800/23794261/36b736fc-0587-11e7-90c0-f371c2144fd3.png)